src/PrettierEditProvider.ts

```diff
@@ -10,41 +10,49 @@ import {
     TextEdit,
     Selection,
     Position
-} from 'vscode';
-import { requireLocalPkg } from './requirePkg';
+} from "vscode";
+
+import { requireLocalPkg } from "./requirePkg";
 
 import {
     PrettierVSCodeConfig,
     PrettierConfig,
     Prettier,
     PrettierEslintFormat
-} from './types.d';
+} from "./types.d";
 type ShowAction = "Show";
 
+import { extname } from "path";
 
 /**
  * Format the given text with user's configuration.
  * @param text Text to format
  * @param path formatting file's path
- * @returns {string} formatted text 
+ * @returns {string} formatted text
  */
 function format(text: string, path: string): string {
-    const config: PrettierVSCodeConfig = workspace.getConfiguration('prettier') as any;
+    const config: PrettierVSCodeConfig = workspace.getConfiguration(
+        "prettier"
+    ) as any;
     /*
     handle deprecated parser option
     */
     let parser = config.parser;
-    if (!parser) { // unset config
-        parser = config.useFlowParser ? 'flow' : 'babylon';
+    if (!parser) {
+        // unset config
+        parser = config.useFlowParser ? "flow" : "babylon";
+    }
+    if (extname(path).startsWith(".ts")) {
```

Comment:
This won't work on an unsaved file. (Untitled)
Could you use [document.languageId](https://code.visualstudio.com/docs/extensionAPI/vscode-api#TextDocument)

You can change format's signature to access the document
Something like:
```js
format(text: string, document: TextDocument): string
```